ReactCommon/jsi/jsi/jsi.h

```diff
@@ -252,6 +252,7 @@ class JSI_EXPORT Runtime {
 
   virtual String createStringFromAscii(const char* str, size_t length) = 0;
   virtual String createStringFromUtf8(const uint8_t* utf8, size_t length) = 0;
+  virtual Object createArrayBufferFromBytes(const void* bytes, size_t length) = 0;
```

Comment:
As @tmikov mentioned, there are multiple implementations of JSI for different JS engines. In addition to JSC and Hermes, there is also V8, Chakra, and a new ABI safe JSI wrapper for Windows. Maybe we could provide a default (non-functional) implementation, and allow JS engine specific implementations to opt-in over time?

```c++
virtual Object createArrayBufferFromBytes(const void* bytes, size_t length) { throw std::runtime_error{"Not implemented."}; }
```

I don't think there is anything like this in JSI already though, so not sure what the right pattern is for expanding API surface area.

Reply:
Hi Ryan, I went ahead and added this. There seems to be precedent for this, which makes me think it's a good suggestion:
https://github.com/facebook/react-native/blob/9c32140068463739b91874689f741ea9630d8c3b/ReactCommon/jsi/JSCRuntime.cpp#L942

I also implemented the same functionality in hermes:
https://github.com/facebook/hermes/pull/419